satviz-sat/src/main/java/edu/kit/satviz/sat/SatAssignment.java

```diff
@@ -1,4 +1,175 @@
 package edu.kit.satviz.sat;
 
+/**
+ * This class represents a finished solution for a SAT-instance.
+ *
+ * @author quorty, LuWae
+ */
 public class SatAssignment {
+
+  /**
+   * This enum represents the variable state.
+   */
+  public enum VariableState {
+    DONTCARE(0),
+    SET(1),
+    UNSET(2),
+    RESERVED(3);
+
+    /**
+     * This contains the 2-bit representation of the variable state (stored as a byte).
+     */
+    public final byte val;
+
+    /**
+     * Simple constructor for mapping byte values to the states.
+     *
+     * @param val Byte representation of the variable state.
+     */
+    VariableState(int val) {
+      this.val = (byte) val;
+    }
+
+    /**
+     * This method converts a given 2-bit state representation into an instance of the
+     * <code>VariableState</code> enum.
+     *
+     * @param val 2-bit representation of the variable state.
+     * @return An instance of the <code>VariableState</code> enum,
+     *         or <code>null</code> (in case <code>val</code> is invalid).
+     */
+    public static VariableState fromValue(int val) {
+      for (VariableState state : VariableState.values()) {
+        if (state.val == val) {
+          return state;
+        }
+      }
+      return null;
+    }
+
+    /**
+     * This method converts a given integer state representation into an instance of the
+     * <code>VariableState</code> enum.
+     *
+     * @param intState Integer representation of the variable state.
+     * @return An instance of the <code>VariableState</code> enum.
+     */
+    public static VariableState fromIntState(int intState) {
+      return switch ((int) Math.signum(intState)) {
+        case 0 -> DONTCARE;
+        case -1 -> UNSET;
+        case 1 -> SET;
+        default -> null;
+      };
+    }
+
+  }
+
+  private final int varCount;
+  private final VariableState[] satAssignment;
+
+  /**
+   * An instance of the SatAssignment class can be used similarly to a
+   * <code>VariableState</code> array.<br>
+   *
+   * <p>
+   * <i>NOTE: The variable index starts with</i> <code>1</code> <i>instead of the usual</i>
+   * <code>0</code>. <code>varCount</code> <i>has to be greater than</i> <code>0</code>.
+   * </p>
+   *
+   * @param varCount The total amount of different variables.
+   */
+  public SatAssignment(int varCount) {
+    if (varCount > 0) {
+      this.varCount = varCount;
+      this.satAssignment = new VariableState[varCount + 1];
+    } else {
+      this.varCount = 0;
+      this.satAssignment = new VariableState[0];
+    }
```

Comment:
Der Sinn dieses `if`s ist mir nicht ganz klar. Warum speichern wir einen int zu viel bei `varCount > 0`? Wir können es einfach wie ein 0-indiziertes array behandeln (intern) und nach außen eine 1-indizierte API anbieten (s.u.).\
Das Attribut `varCount` können wir uns auch sparen; Das ist dann einfach `satAssignment.length`.
```suggestion
    this.satAssignment = new VariableState[varCount];
```

Reply:
Der array wird nun intern wie ein 0-indiziertes array behandelt und es werden nun IllegalArgumentExceptions ausgegeben im falle von ungültigen Parametern!